query-engine/core/src/query_graph_builder/write/update.rs

```diff
@@ -342,3 +358,9 @@ fn can_use_atomic_update(
 
     true
 }
+
+pub struct UpdateManyRecordNodeOptionals<'a> {
```

Comment:
Linter was complaining about too many params on the `update_many_record_node` method so I extracted the most recently introduced optional params into this param object.

Happy to learn if you have a stylistic better or more idiomatic approach for this. :)

Reply:
I think it's okay, but I think a slightly more idiomatic approach would be to refactor `update_many_record_node` to a builder pattern (something like `UpdateManyRecordNodeBuilder::new(graph, query_schema, filter, model, data_map).limit(1).build()`).